devito/finite_differences/derivative.py

```diff
@@ -207,6 +206,36 @@ def evaluate(self):
             res = generic_derivative(expr, *self.dims, self.fd_order,
                                      self.deriv_order, stagger=self.stagger[0],
                                      matvec=self.transpose)
-        for e in self._eval_at:
-            res = res.xreplace(e)
         return res
+
+
+class Eval(Derivative):
+    """
+    Class that represent unevaluated Derivative at a point. This class
+    is used to delay evaluation of subs and xreplace of Derivative and is
+    particularly useful for SparseFunctions injection and interpolations
+    that involve derivatives.
+    
+    """
+    def __new__(cls, expr, variables, point, **kwargs):
+        variables = as_tuple(variables)
+        point = as_tuple(point)
+
+        obj = Derivative.__new__(cls, expr.expr, *expr.dims, deriv_order=expr.deriv_order,
```

Comment:
I think this makes no sense. ``Eval`` should wrap ``Derivative`` (or any another evaluable object), *not become* a Derivative itself.

Also, the fact that you're adding attributes that you don't have when you create a Derivative (e.g., ``_rule``) is a hack. Instead, if Eval is a first-class citizen, it's absolutely fine to keep ``_rule`` around